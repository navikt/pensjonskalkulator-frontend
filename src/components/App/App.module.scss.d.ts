import globalClassNames from '../../style.d'
declare const classNames: typeof globalClassNames & {
  readonly main: 'main'
}
export = classNames
