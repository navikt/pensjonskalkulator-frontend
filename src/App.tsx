import { useState } from 'react'

import { Button } from '@navikt/ds-react'

import reactLogo from './assets/react.svg'

import styles from './App.module.scss'

function App() {
  const [count, setCount] = useState(0)

  return (
    <div className="App">
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img
            src="/pensjon/kalkulator/vite.svg"
            className={styles.logo}
            alt="Vite logo"
          />
        </a>
        <a href="https://reactjs.org" target="_blank">
          <img
            src={reactLogo}
            className={`${styles.logo} ${styles['logo--react']}`}
            alt="React logo"
          />
        </a>
      </div>
      <h1 className={styles.title}>Pensjonskalkulator</h1>
      <div className={styles.card}>
        <Button
          className={styles.button}
          onClick={() => setCount((c) => c + 1)}
        >
          count is {count}
        </Button>
        <p>Lorem ipsum dolor sit amet</p>
      </div>
    </div>
  )
}

export default App
