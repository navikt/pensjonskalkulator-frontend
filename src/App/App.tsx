import React from 'react'
import clsx from 'clsx'

import frameStyles from '../Frame/Frame.module.scss'
import { Pensjonsberegning } from '../Pensjonsberegning'

import styles from './App.module.scss'

export function App() {
  return (
    <main
      className={clsx(
        frameStyles.frame,
        frameStyles.frame_isFlex,
        frameStyles.frame_hasPadding,
        styles.main
      )}
    >
      <Pensjonsberegning/>
    </main>
  )
}
