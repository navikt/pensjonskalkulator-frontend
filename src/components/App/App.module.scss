@import '../../scss/variables';

.main {
  padding-top: var(--a-spacing-6);
  display: flex;
  align-items: start;
}

.headerGroup {
  padding: 0 var(--a-spacing-6);
}

.section {
  display: flex;
  flex-direction: column;
  gap: var(--a-spacing-4);
  padding: var(--a-spacing-6);
  text-align: start;
  margin-bottom: var(--a-spacing-6);

  @media (min-width: $a-breakpoint-sm) {
    padding: var(--a-spacing-10) var(--a-spacing-10);
  }
}
