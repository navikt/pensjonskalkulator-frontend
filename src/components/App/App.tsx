import React from 'react'

import { Heading } from '@navikt/ds-react'

import { TidligstMuligeUttak } from '@/components/TidligstMuligeUttak'

import styles from './App.module.scss'

export function App() {
  return (
    <main className={styles.main}>
      <div className={styles.headerGroup}>
        <Heading size="large" level="1" spacing>
          Pensjonskalkulator
        </Heading>
      </div>
      <TidligstMuligeUttak />
    </main>
  )
}
