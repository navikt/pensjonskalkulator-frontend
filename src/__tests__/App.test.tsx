import renderer, { ReactTestRenderer } from 'react-test-renderer'

import { describe, vi, expect, test } from 'vitest'

import App from '../App'
import {
  act,
  toJson,
  render,
  userEvent,
  screen,
  createSuccessFetchResponse,
  createFailureFetchResponse,
} from '../test-utils'

const data = require('../__mocks__/data.json')

function mockLivenessSuccess() {
  const fetchMock = vi.fn().mockResolvedValue(createSuccessFetchResponse(data))
  global.fetch = fetchMock
}

function mockLivenessFailure() {
  const fetchMock = vi.fn().mockResolvedValue(createFailureFetchResponse())
  global.fetch = fetchMock
}

function mockConsole() {
  const consoleMock = vi.fn()
  global.console.warn = consoleMock
}

describe('Gitt at appen importeres,', () => {
  test('Når Appen starter, Så lages det en GET request til liveness status', async () => {
    mockLivenessSuccess()
    await act(async () => {
      renderer.create(<App />)
    })
    expect(fetch).toHaveBeenCalledWith('/internal/health/liveness', {
      method: 'GET',
    })
  })

  test('Når Appen starter og at requesten er vellykket, Så rendres den slik den skal', async () => {
    mockLivenessSuccess()
    const component = renderer.create(<App />)
    await act(async () => {
      const tree = toJson(component)
      expect(tree).toMatchSnapshot()
    })
  })

  test('Når Appen starter og at requesten feiler, Så logges feilmeldingen og appen vises uten liveness status', async () => {
    mockLivenessFailure()
    mockConsole()
    let component
    await act(async () => {
      component = renderer.create(<App />)
    })
    expect(fetch).toHaveBeenCalled()
    expect(console.warn).toHaveBeenCalledWith(new Error('Something went wrong'))
    const tree = toJson(component as unknown as ReactTestRenderer)
    expect(tree).toMatchSnapshot()
  })
})

test('Når brukeren klikker på count knappen, Så øker count', async () => {
  render(<App />)
  const btn = screen.getByRole('button')
  expect(btn.textContent).toBe('count is 0')
  await userEvent.click(btn)
  expect(btn.textContent).toBe('count is 1')
})
