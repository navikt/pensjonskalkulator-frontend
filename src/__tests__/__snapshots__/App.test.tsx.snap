// Vitest Snapshot v1

exports[`Gitt at appen importeres, > Når Appen starter og at requesten er vellykket, Så rendres den slik den skal 1`] = `
<div
  className="App"
>
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        className="_logo_ce1e69"
        src="/pensjon/kalkulator/vite.svg"
      />
    </a>
    <a
      href="https://reactjs.org"
      target="_blank"
    >
      <img
        alt="React logo"
        className="_logo_ce1e69 _logo--react_ce1e69"
        src="/pensjon/kalkulator/src/assets/react.svg"
      />
    </a>
  </div>
  <h1
    className="_title_ce1e69"
  >
    Pensjonskalkulator is "
    "
  </h1>
  <div
    className="_card_ce1e69"
  >
    <button
      className="_button_ce1e69 navds-button navds-button--primary navds-button--medium"
      onClick={[Function]}
      style={
        {
          "width": undefined,
        }
      }
    >
      <span
        aria-live="polite"
        className="navds-label"
      >
        count is 
        0
      </span>
    </button>
    <p>
      Du bør muligens spare bittelitt mer altså...
    </p>
  </div>
</div>
`;

exports[`Gitt at appen importeres, > Når Appen starter og at requesten feiler, Så logges feilmeldingen og appen vises uten liveness status 1`] = `
<div
  className="App"
>
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        className="_logo_ce1e69"
        src="/pensjon/kalkulator/vite.svg"
      />
    </a>
    <a
      href="https://reactjs.org"
      target="_blank"
    >
      <img
        alt="React logo"
        className="_logo_ce1e69 _logo--react_ce1e69"
        src="/pensjon/kalkulator/src/assets/react.svg"
      />
    </a>
  </div>
  <h1
    className="_title_ce1e69"
  >
    Pensjonskalkulator is "
    "
  </h1>
  <div
    className="_card_ce1e69"
  >
    <button
      className="_button_ce1e69 navds-button navds-button--primary navds-button--medium"
      onClick={[Function]}
      style={
        {
          "width": undefined,
        }
      }
    >
      <span
        aria-live="polite"
        className="navds-label"
      >
        count is 
        0
      </span>
    </button>
    <p>
      Du bør muligens spare bittelitt mer altså...
    </p>
  </div>
</div>
`;
